app/scripts/controllers/rif/rns/resolver/index.js

```diff
@@ -131,27 +142,60 @@ export default class RnsResolver extends RnsJsDelegate {
    * @param resolverAddress Address of the new resolver to be setted
    * @returns {Promise<unknown>}
    */
-  setResolver (domainName, resolverAddress) {
+  setResolver (domainName, resolverAddress, subdomain = '') {
     return new Promise((resolve) => {
       const domain = this.getDomain(domainName);
       if (domain) {
-        domain.pendingActions.pendingSetResolver = true;
+        if (subdomain) {
+          domain.subdomains.forEach(subdomainItem => {
+            if (subdomainItem.name === subdomain) {
+              subdomainItem.pendingSetResolver = true
+            }
+          });
+        } else {
+          domain.pendingActions.pendingSetResolver = true;
+        }
         this.updateDomain(domain);
       }
-      const transactionListener = this.send(this.rnsContractInstance, 'setResolver', [namehash.hash(domainName), resolverAddress]);
+      let node = namehash.hash(domainName);
+      if (subdomain) {
+        node = rskNameHash(domainName);
```

Comment:
I think this can be an utilitary function called buildSubdomainName or something, since its replicated twice at least

Reply:
You're right, will move it to a function